### PR TITLE
feat: add support for global data source storage

### PR DESCRIPTION
**Pull request by eoaksnes**
## What does this pull request change?

Add support for "global folders" by configuring it in the data source files like this:

```
{
  "name": "SomeDataSource",
  "repositories": {
    ....
  },
  "global_folders": [
    "global"
  ]
}
```

This will make the files inside the "global" folders that contain global files. 

It is possible to reference the global files on disk like this.

```
{
    "type": "CORE:Reference",
    "address": "/global/<filename>.json",
    "referenceType": "storage"
 }
```

The address will upon uploading be replaced with the global id that the document got during import.

### CODE REVIEW

File: dm_cli/command_group/data_source.py [hunk 1]
```diff
@@ -114,21 +117,37 @@ def import_data_source_file(
...
+            for root_package in root_packages:
+                dmss_exception_wrapper(remove_by_path_ignore_404, f"/{data_source_name}/{root_package.name}")
+
+            global_files_storage = io.BytesIO()
+            with ZipFile(global_files_storage, mode="w") as global_files:
+                # Zip all files from the global folders  for easy access to these files later on.
+                # If there will be any performance issues, we should instead use repository pattern.
+                for global_folder in global_folders:
+                    if not Path(data_source_data_dir / global_folder).is_dir():
+                        raise Exception(f"The global folder does not exist '{global_folder}'")
```
Comment (soofstad) [inline dm_cli/command_group/data_source.py:135]: Does this really need to raise an exception?
IMO, if there is no "global", that is fine.

File: dm_cli/package_tree_from_zip.py [hunk 2]
```diff
@@ -68,7 +68,7 @@ def package_tree_from_zip(
                 continue
             if Path(filename).suffix != ".json":
                 file_like = io.BytesIO(zip_file.read(f"{folder_name}/{filename}"))
-                print(f"\tAdding file {filename} ({file_like.getbuffer().nbytes} bytes)")
+                # print(f"\tAdding file {filename} ({file_like.getbuffer().nbytes} bytes)")
```
Comment (soofstad) [inline dm_cli/package_tree_from_zip.py:71]: Just remove completely?

File: dm_cli/bin/dm [hunk 3]
```diff
@@ -96,7 +96,7 @@ def export_entity(target: Annotated[str, typer.Argument(help="Address to the ent
 @app.command("reset")
 def reset(path: Annotated[str, typer.Argument(help="The path on local file system to the data source you want to reset.")],
           validate_entities: Annotated[bool, typer.Option(help="if True, all entities uploaded to DMSS will be validated.")] = True,
-          resolve_local_ids: Annotated[bool, typer.Option(help="if True, will resolve all local ids found in all entities")] = True):
+          resolve_local_ids: Annotated[bool, typer.Option(help="if True, will resolve all local ids found in all entities")] = False):
```
Comment (collinlokken) [inline dm_cli/bin/dm:99]: 😄

File: dm_cli/package_tree_from_zip.py (hunk 2 above)
Comment (collinlokken) [inline dm_cli/package_tree_from_zip.py:71]: remove?

**Review by collinlokken — APPROVED** [review]
Looks good!

File: tests/test_data/test_app_dir_struct/data_sources/DemoApplicationDataSource.json [hunk 4]
```diff
@@ -14,5 +14,8 @@
         "blob"
       ]
     }
-  }
+  },
+  "global_folders": [
+    "global"
+  ]
```
Comment (soofstad) [inline tests/test_data/test_app_dir_struct/data_sources/DemoApplicationDataSource.json:20]: Could this be just an implicit thing?
So the user don't have to think about where this is and what it's called. Perhaps call it "_global_"

**Review by soofstad — CHANGES_REQUESTED** [review]

File: dm_cli/command_group/data_source.py (hunk 1 above)
Comment (eoaksnes) [reply dm_cli/command_group/data_source.py]: Good idea, I changed it so that it just checks that the folder exists, instead of raising an exception.

**Pull request merged**